app/models/mixins/purging_mixin.rb

```diff
@@ -41,23 +39,43 @@ def purge_count_by_date(older_than)
       purge_conditions(older_than).count
     end
 
-    def purge_by_date(older_than, window, &block)
-      _log.info("Purging records older than [#{older_than}]...")
-      total = purge_in_batches(purge_conditions(older_than), window, &block)
-      _log.info("Purging records older than [#{older_than}]...Complete - Deleted #{total} records")
+    def purge_by_date(older_than, window = nil, &block)
+      _log.info("Purging #{table_name.humanize} older than [#{older_than}]...")
+      total = purge_in_batches(purge_conditions(older_than), window || purge_window_size, &block)
+      _log.info("Purging #{table_name.humanize} older than [#{older_than}]...Complete - Deleted #{total} records")
+      total
     end
 
-    def purge_in_batches(conditions, window, total = 0)
-      query = conditions.limit(window)
+    def purge_in_batches(conditions, window, total = 0, total_limit = nil)
+      query = conditions.select(:id).limit(window)
```

Comment:
I'm confused.  I thought you *didn't* want to pull back ids, and the original implementation did just that, yet this had been change to select the ids?  Or is it using a subquery?  I guess I'm confused how this is better.